tests/nextflowtower/test_client.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os
+from types import GeneratorType
+
+import pytest
+
+from sagetasks.nextflowtower import client
+
+EG_TOKEN = "token"
+EG_API_URL = "https://example.com"
+EG_METHOD = "GET"
+EG_ENDPOINT = "/example/endpoint"
+
+
+@pytest.fixture
+def tower_client():
+    return client.TowerClient(EG_TOKEN, EG_API_URL)
+
+
+def test_import():
+    assert client
+
+
+class TestTowerClient:
+    def test_missing_args(self):
+        # Clear environment
+        env_vars = [
+            "NXF_TOWER_TOKEN",
+            "TOWER_ACCESS_TOKEN",
+            "NXF_TOWER_API_URL",
+            "TOWER_API_ENDPOINT",
+        ]
+        for var in env_vars:
+            if var in os.environ:
+                del os.environ[var]
+        # Ensure exception when initializing
+        with pytest.raises(KeyError):
+            client.TowerClient()
+
+    def test_get_valid_name(self, tower_client):
+        expectations = {
+            "foobar": "foobar",
+            "FooBar": "FooBar",
+            "foo-bar": "foo-bar",
+            "foo_bar": "foo_bar",
+            "f00b4r": "f00b4r",
+            "foo,bar": "foo-bar",
+            "foo (bar)": "foo--bar-",
+        }
+        for name, output in expectations.items():
+            assert tower_client.get_valid_name(name) == output
+
+    def test_request(self, mocker, capfd, tower_client):
```

Comment:
I like the use of pytest-mock here.  Even though it adds a dependency, it simplifies the code a bit by adding the mocker fixture

Reply:
It also eliminates the need of ugly multiline `with patch.object(...)` statements like [this one](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/ccf67ca9991165ed2f5522c8faa6e1ef5028742d/tests/unit/synapseutils/unit_test_synapseutils_sync.py#L39-L45) and avoids unnecessary indentation since `pytest-mock` automatically undoes the mocking at the end of each test.